Gordon360/Services/ProfileService.cs

```diff
@@ -88,6 +88,29 @@ public MailboxViewModel GetMailboxCombination(string username)
             return combo;
         }
 
+        /// <summary>
+        /// get the current user's birthday
+        /// </summary>
+        /// <param name="username">The current user's username</param>
+        /// <returns>Date the user's date of birth</returns>
+        public DateTime GetBirthday(string username)
+        {
+            var dob = _unitOfWork.AccountRepository.FirstOrDefault(a => a.AD_Username == username).Birth_Date;
+
+            if (dob == null)
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "A birthday was not found for this user." };
+            }
+
+            try
+            {
+                return (DateTime)(dob);
+            } catch
+            {
+                throw new ResourceNotFoundException() { ExceptionMessage = "A birthday was not found for this user." };
```

Comment:
Technically, the error this is catching would be for converting the birthdate to a DateTime. You've already handled the not found case above.